R/data_parsing.R

```diff
@@ -207,19 +207,27 @@ split_dbnsfp_values <- function(variant_dataframe) {
     "PROVEAN_pred"
   )
   
-  # "&"-splits
-  etsplits <- lapply(X = variant_dataframe[, process_columns], FUN = function(x) stringr::str_split(string = x, pattern = fixed("&")))
+  # "&"-splits.
+  # Note that the output is a list that contains columns (as lists) that contains rows, 
+  # each of which is now a vector, since it may contain multiple values that were separated by '&'.
+  # etsplits = list( col1 = list( row_1 = c("val1", "val2", ...), ... ), ... )
+  etsplits <- lapply(X = variant_dataframe[, process_columns, drop = FALSE], 
+                     FUN = function(x) stringr::str_split(string = x, pattern = fixed("&")))
   
-  match_indices <- mapply(variant_dataframe$Feature,
-                          etsplits$Ensembl_transcriptid,
-                          FUN = match)
+  # For each row, try to match the transcript given by VEP (Feature-column)
+  # to the vector of transcripts given by dbNSFP (Ensembl_transcriptid-column).
+  # Output is a list whose elements are vectors that contain every match for the row.
+  match_indices <- mapply(FUN = match,
+                          x = variant_dataframe$Feature,
+                          table = etsplits$Ensembl_transcriptid)
   
-  # Choose the value for the transcript
   picked_values <- lapply(X = etsplits, 
                           FUN = function(column) 
                             mapply(
-                              match_indices, column,  
-                              FUN = function(index, col) if (!is.na(index)) col[index] else NA 
```

Comment:
This function was redundant to define, since the operator `[` already returns NA if the index is NA.